src/lib/fillEvent.ts

```diff
@@ -0,0 +1,187 @@
+
+export const getStorageCylinder = async (
+  trx: Knex.Transaction,
+  id: number
+): Promise<StorageCylinder> => {
+  const storageCylinderQuery = await trx<StorageCylinder>('storage_cylinder')
+    .where('id', id)
+    .first('id', 'gas_id as gasId', 'volume', 'name');
+  if (storageCylinderQuery === undefined) {
+    log.error('Storage cylinder not found');
+    throw new Error('Storage cylinder not found');
+  }
+  const sc: StorageCylinder = JSON.parse(JSON.stringify(storageCylinderQuery));
+  return sc;
+};
+
+const getActivePriceId = async (
+  trx: Knex.Transaction,
+  gasId: number
+): Promise<number> => {
+  const prices: GasPrice[] = await trx<GasPrice>('gas_price')
+    .where('gas_id', gasId)
+    .andWhere('active_to', '>', knexController.fn.now())
+    .andWhere('active_from', '<', knexController.fn.now())
+    .select('id');
+  const pricesArr = prices.map((price) => JSON.parse(JSON.stringify(price)));
+  if (pricesArr.length > 1) {
+    log.error(`Multiple active prices were found for gasId: ${gasId}`);
+    throw new Error(`Multiple active prices`);
+  } else if (prices.length === 0) {
+    log.error(`No price was found for gasId: ${gasId}`);
+    throw new Error(`Price not found`);
+  }
+  return pricesArr[0].id;
+};
+
+const getAirGasId = async (trx: Knex.Transaction): Promise<number> => {
+  const air: Gas = await trx<Gas>('gas').where('name', 'Air').first('id');
+  if (air === undefined) {
+    log.error('Gas id was not found for air');
+    throw new Error('Gas id was not found for air');
+  }
+  return air.id;
+};
+
+export const createFillEvent = async (
+  authUser: AuthUser,
+  body: CreateFillEventBody
+): Promise<{ status: number; message: string; fillEventId?: number }> => {
+  const {
+    cylinderSetId,
+    gasMixture,
+    filledAir,
+    storageCylinderUsageArr,
+    description,
+  } = body;
+
+  if (!filledAir && storageCylinderUsageArr.length === 0) {
+    return { status: 400, message: 'No gases were given' };
+  }
+
+  const trx = await knexController.transaction();
+  const userQuery = await trx<User>('user')
+    .where('id', authUser.id)
+    .first('id', 'is_blender as isBlender');
+
+  const user: User = JSON.parse(JSON.stringify(userQuery));
+
+  if (!user.isBlender && storageCylinderUsageArr.length !== 0) {
+    await trx.rollback();
+    return { status: 403, message: 'User does not have blender priviledges' };
+  }
+  const set = selectCylinderSet(trx, cylinderSetId);
+  if (set === undefined) {
+    await trx.rollback();
+    return { status: 400, message: 'Cylinder set was not found' };
+  }
+  const params = [user.id, cylinderSetId, gasMixture];
+  let sql: string;
+  if (description === undefined) {
+    sql =
+      'INSERT INTO fill_event (user_id, cylinder_set_id, gas_mixture) VALUES (?,?,?) RETURNING id';
+  } else {
+    sql =
+      'INSERT INTO fill_event (user_id, cylinder_set_id, gas_mixture, description) VALUES (?,?,?,?) RETURNING id';
+    params.push(description);
+  }
```

Comment:
```suggestion
const sql = `INSERT INTO fill_event (user_id, cylinder_set_id, gas_mixture ${ description ? ', description' : '' }) VALUES (?,?,?, ${ description ? ',?' : '' }) RETURNING id`;
```

Tai sitten voit tehdä 
```
const fields = ['user_id, ...'];
if (description) fields.push('description');

const sql = '...';
```

Reply:
Näistä alempi on varmaan siistimpi =)

Reply:
Jos descriptioniksi lähettää NULL arvon tilantessa että `description === undefined` niin sillon ton ekan ternaryn vois jättää pois?

Reply:
Tein tuon omasta mielestä siististi (ja disabloin eslintin siltä riviltä 🌚)